tools/donate-cpu.py

```diff
@@ -255,25 +269,27 @@ def scanPackage(workPath, cppcheckPath, jobs):
         if os.path.exists(os.path.join(cppcheckPath, 'cfg', library + '.cfg')) and hasInclude('temp', includes):
             libraries += ' --library=' + library
 
-# Reference for GNU C: https://gcc.gnu.org/onlinedocs/cpp/Common-Predefined-Macros.html
+    # Reference for GNU C: https://gcc.gnu.org/onlinedocs/cpp/Common-Predefined-Macros.html
     options = jobs + libraries + ' -D__GNUC__ --check-library --inconclusive --enable=style,information --platform=unix64 --template=daca2 -rp=temp temp'
-    cmd = 'nice ' + cppcheckPath + '/cppcheck' + ' ' + options
-    print(cmd)
-    startTime = time.time()
-    p = subprocess.Popen(cmd.split(), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-    comm = p.communicate()
-    stopTime = time.time()
-    stdout = comm[0].decode(encoding='utf-8', errors='ignore')
-    stderr = comm[1].decode(encoding='utf-8', errors='ignore')
-    if p.returncode != 0 and 'cppcheck: error: could not find or open any of the paths given.' not in stdout:
-        # Crash!
-        print('Crash!')
-        return -1, '', '', -1, options
-    if stderr.find('Internal error: Child process crashed with signal 11 [cppcheckError]') > 0:
+    cppcheck_cmd = cppcheckPath + '/cppcheck' + ' ' + options
+    cmd = 'nice ' + cppcheck_cmd
+    returncode, stdout, stderr, elapsedTime = runCommand(cmd)
+    if returncode == -11 or stderr.find('Internal error: Child process crashed with signal 11 [cppcheckError]') > 0:
         # Crash!
         print('Crash!')
-        return -1, '', '', -1, options
-    elapsedTime = stopTime - startTime
+        stacktrace = ''
+        if cppcheckPath == 'cppcheck':
+            # re-run within gdb to get a stacktrace
+            cmd = 'gdb --batch --eval-command=run --eval-command=bt --return-child-result --args ' + cppcheck_cmd + " -j1"
+            returncode, stdout, stderr, elapsedTime = runCommand(cmd)
+            gdb_pos = stdout.find("Program received signal")
```

Comment:
I have tested it under Cygwin and it does not print the word "Program" but the actual program name "cppcheck" (See the uploaded result for http://cppcheck.osuosl.org:8000/scram).
One solution would be to simply remove the word "Program" here:
```suggestion
            gdb_pos = stdout.find(" received signal")
```

Reply:
In case there is no "Checking" output that would cause the output to be truncated. But I think we can leave it like this until we have to cross that bridge.